draft-ietf-privacypass-integration.md

```diff
@@ -0,0 +1,224 @@
+{{I-D.irtf-cfrg-voprf, Section 4}}, and KeyConfig.public_key corresponds
+to a serialized public key of length `Ne` bytes (denoted as a
+`SerializedElement` in {{I-D.irtf-cfrg-voprf, Section 2}}). In
+
+In order for higher-level applications to indicate which key
+configuration is being used, a common identifier, such as
+`id=SHA256(KeyConfig)`, should be used. Note that the ciphersuite that
+is used is determined entirely by the choice of `suite` in the server
+key configuration.
+
+## Issuance phase
+
+Let `info` be the agreed upon metadata between client and server, and
+let `config` be the server's chosen key configuration.
+
+First, a client configures its verifiable context using `config`:
+
+~~~
+client_context = SetupVerifiableClient(
+                  config.suite, config.public_key
+                 )
+~~~
+
+Likewise, the server creates its own context using `config` and the
+corresponding private key `key`:
+
+~~~
+server_context = SetupVerifiableServer(
+                  config.suite, key, config.public_key
+                 )
+~~~
+
+The client then creates an issuance request for a random value `nonce`
+as follows:
+
+~~~
+nonce = random(32)
+blind, blindedElement = context.Blind(nonce)
+~~~
+
+The client then sends `blindedElement` to the server. The server, upon
+receipt, evaluates the request:
+
+~~~
+evaluatedElement, proof = server_context.Evaluate(
+                           key, config.public_key,
+                           blindedElement, info
+                          )
+~~~
+
+The server sends both `evaluatedElement` and `proof` to the client.
+These are concatenated together. As the length of both is fixed, there
+is no ambiguity in parsing the result.
+
+The client then completes issuance as follows:
+
+~~~
+output = client_context.Finalize(
+          nonce, blind, evaluatedElement, info
+         )
+~~~
+
+This procedure may fail with an error (`VerifyError` or
+`DeserializeError`), in which case the issuance is said to have failed.
+The output of the issuance protocol is the concatenation of `nonce` and
+`output`, denoted as `token`:
+
+~~~
+struct {
+   uint8 nonce[32];
+   uint8 output[Nh];
+} Token;
+~~~
+
+where `Nh` is as defined in {{I-D.irtf-cfrg-voprf}}.
+
+## Redemption phase
+
+The client sends the `Token` to the server to verify locally. In
+particular, the server verifies the `Token` as follows:
+
+~~~
+valid = server_context.VerifyFinalize(
+         key, token.nonce, token.output, info
+        )
+~~~
+
+Redemption is considered successful if `valid` is true.
+
+# IANA Considerations
+
+This specification registers new ciphersuites for the Privacy Pass
+protocol, corresponding to the following ciphersuites detailed in
+{{I-D.irtf-cfrg-voprf}}.
+
+- Ciphersuite OPRF(ristretto255, SHA-512) = XXXX
+- Ciphersuite OPRF(decaf448, SHAKE-256) = XXXX
+- Ciphersuite OPRF(P-256, SHA-256) = XXXX
+- Ciphersuite OPRF(P-384, SHA-384) = XXXX
+- Ciphersuite OPRF(P-521, SHA-512) = XXXX
```

Comment:
Assuming we want to do a separate IANA registry, we might want to pare down the ones that are initially defined?